eduhelx_jupyterlab_student/student_repo.py

```diff
@@ -65,3 +71,19 @@ def _compute_current_assignment(assignments, repo_root, current_path):
                 break
 
         return current_assignment
+
+    @classmethod
+    def from_assignment_no_path(cls, course, assignments, assignment_id: int):
+        try:
+            assignment = [a for a in assignments if a["id"] == assignment_id][0]
+        except IndexError:
+            raise NotInAnAssignmentException
+        
+        repo_root = cls._compute_repo_root(course["name"]).resolve()
+        assignment_path = repo_root / assignment["directory_path"]
```

Comment:
This can use `get_assignment_path(assignment)`

Reply:
get_assignment_path is not static